RiotSwiftUI/Modules/UserSessions/DeviceAvatar/DeviceAvatarView.swift

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+import DesignKit
+
+/// Avatar view for device
+struct DeviceAvatarView: View {
+    
+    @Environment(\.theme) var theme: ThemeSwiftUI
+    
+    var viewData: DeviceAvatarViewData
+        
+    var avatarSize: Float = 40.0
+    var badgeSize: Float {
+        return 24
+    }
+    
+    var body: some View {
+        ZStack {
+            VStack {
+                VStack(alignment: .center) {
+                    viewData.deviceType.image
+                }
+                .padding()
+            }
+            .frame(maxWidth: CGFloat(avatarSize), maxHeight: CGFloat(avatarSize))
+            .background(theme.colors.system)
+            .clipShape(Circle())
+            
+            if let isVerified = viewData.isVerified {
+                
+                VStack(alignment: .trailing) {
+                    Spacer()
+                    HStack(alignment: .bottom) {
+                        Spacer()
+                        VStack() {
```

Comment:
You can remove these 3 stacks by changing the `ZStack` to have an alignment:
```
ZStack(alignment: .bottomTrailing) { …
```

Reply:
Good catch!